commerce_attribute.module

```diff
@@ -0,0 +1,122 @@
+<?php
+
+/**
+ * @file
+ * Hooks and functionality for the Commerce Attribute module.
+ */
+
+use Drupal\entity\BundleFieldDefinition;
+use Drupal\field\Entity\FieldStorageConfig;
+use Drupal\Core\Form\FormStateInterface;
+
+/**
+ * Implements hook_entity_type_alter().
+ */
+function commerce_attribute_entity_type_alter(&$entity_types) {
+  $entity_types['commerce_product_attribute']->setClass('Drupal\commerce_attribute\Entity\ProductAttribute');
+  $entity_types['commerce_product_attribute_value']->setClass('Drupal\commerce_attribute\Entity\ProductAttributeValue');
+  $entity_types['commerce_product_attribute']->setFormClass('add', 'Drupal\commerce_attribute\Form\ProductAttributeForm');
+  $entity_types['commerce_product_attribute']->setFormClass('edit', 'Drupal\commerce_attribute\Form\ProductAttributeForm');
+  $entity_types['commerce_product_attribute_value']->setFormClass('add', 'Drupal\commerce_attribute\Form\ProductAttributeValueForm');
+  $entity_types['commerce_product_attribute_value']->setFormClass('edit', 'Drupal\commerce_attribute\Form\ProductAttributeValueForm');
+}
+
+/**
+ * Implements hook_ENTITY_TYPE_update().
+ */
+function commerce_attribute_commerce_product_attribute_value_update($attribute_value) {
+
+  // Proceed only if price trait is enabled.
+  if (!$attribute_value->hasField('price')) {
+    return;
+  }
+  // Check if the attribute price is changed or not.
+  // If its not changed, we do not have to perform the variation
+  // price update process.
+  $attribute_value_original = $attribute_value->original;
+  $original_price = $attribute_value_original->get('price')->first()->toPrice()->getNumber();
+  $new_price = $attribute_value->get('price')->first()->toPrice()->getNumber();
+  if ($new_price == $original_price) {
+    return;
+  }
```

Comment:
**CHANGE REQUEST**

*Performance/Readability*

The `$attribute_value_original` is only used to the get the price before the update. It does not contribute to readability i.e. `$attribute_value_original` is not that much more readable than `$attribute_value->original`, and there is no readability issue here in the first place. So there's no reason to allocate an unnecessary variable here.

*API use*

Then, let's use the Price API that provides functions for comparing prices.

With some refactoring to avoid code repetition and respecting line length soft limit, the above changes would be:

```
$original_price = $this->getPrice($attribute_value->original);
$new_price = $this->getPrice($attribute_value);
if (!$new_price->compareTo($original_price)) {
  return;
}
```
and
```
protected function getPrice(ProductAttributeValueInterface $value) {
  return $value->get('price')->first()->toPrice();
}
```

Reply:
I have modified the code. As this is in a module file, I do not think we can add protected functions. Do we need to define getPrice in a service class?